corporate/lib/stripe.py

```diff
@@ -622,6 +658,78 @@ def is_sponsored(self) -> bool:
     def is_sponsored_or_pending(self, customer: Optional[Customer]) -> bool:
         pass
 
+    @abstractmethod
+    def get_sponsorship_request_session_specific_context(
+        self,
+    ) -> SponsorshipRequestSessionSpecificContext:
+        pass
+
+    @abstractmethod
+    def save_org_type_from_request_sponsorship_session(self, org_type: int) -> None:
+        pass
+
+    def request_sponsorship(self, form: SponsorshipRequestForm) -> None:
+        if not form.is_valid():
+            message = " ".join(
+                error["message"]
+                for error_list in form.errors.get_json_data().values()
+                for error in error_list
+            )
+            raise BillingError("Form validation error", message=message)
+
+        request_context = self.get_sponsorship_request_session_specific_context()
+        with transaction.atomic():
+            # Ensures customer is created first before updating sponsorship status.
+            self.update_customer_sponsorship_status(True)
+            sponsorship_request = ZulipSponsorshipRequest(
+                customer=self.get_customer(),
+                requested_by=request_context["realm_user"],
+                org_website=form.cleaned_data["website"],
+                org_description=form.cleaned_data["description"],
+                org_type=form.cleaned_data["organization_type"],
+                expected_total_users=form.cleaned_data["expected_total_users"],
+                paid_users_count=form.cleaned_data["paid_users_count"],
+                paid_users_description=form.cleaned_data["paid_users_description"],
+            )
+            sponsorship_request.save()
+
+            org_type = form.cleaned_data["organization_type"]
+            self.save_org_type_from_request_sponsorship_session(org_type)
+
+            if request_context["realm_user"] is not None:
+                # TODO: Refactor to not create an import cycle.
+                from zerver.actions.users import do_change_is_billing_admin
+
+                do_change_is_billing_admin(request_context["realm_user"], True)
+
+            org_type_display_name = get_org_type_display_name(org_type)
+
+        # TODO: Refactor to not create an import cycle.
+        from corporate.lib.support import get_support_url
+
+        user_info = request_context["user_info"]
+        support_url = get_support_url(request_context["realm_string_id"])
```

Comment:
Reading this now, I think we just need to make `self.get_support_url` a function on `BillingSession`; in particular, it shouldn't involve a `realm_string_id` for the other session types.

Reply:
Made this change and dropped the `get_support_url` parameters change. These URLs will have a different form anyway, so it has to be a more core function of the `BillingSession` type. @laurynmm FYI.